templates/template_modbus.go

```diff
@@ -0,0 +1,104 @@
+package templates
+
+import (
+	"bytes"
+	_ "embed"
+	"errors"
+	"regexp"
+	"strings"
+	"text/template"
+
+	"github.com/Masterminds/sprig/v3"
+	"github.com/thoas/go-funk"
+)
+
+//go:embed modbus.tpl
+var modbusTmpl string
+
+func (t Template) RenderModbusTemplate(values map[string]interface{}, indentlength int) (string, error) {
```

Comment:
Der Teil sieht hacky aus. Was soll hier passieren?

Reply:
Hier werden die, je nach wert von `modbus`(also `rs485serial`, `rs485tcpip` oder `tcpip`), passenden modbus spezifischen Parameter, die der Benutzer angegeben hat, an der "richtigen" Stelle (Einrücken) in der Hierarchie der Konfiguration hinzugefügt. Damit das rendern und das instanziieren dann auch funktioniert.

Reply:
puh. Könnten wir das nicht durch eingebundene Templates "schöner" lösen?

Reply:
Das wird in den Templates ja mit dem magic comment gelöst. Und damit hast diese ganzen Cases nicht x-fach in vielen Templates drin und damit auch Fehleranfälligkeit.

Reply:
Beispiel: https://github.com/evcc-io/evcc/pull/1888/files#diff-a8d84805f72971c964a1331f6aac5fbaf867ac015bc9eaac7140a94c24f8ce99

Reply:
Du müsstest den Inhalt von modbus.tpl dann an jede Stelle in jedes Template reinkopieren. Imho nicht der richtige Weg

Reply:
Du kannst Templates in Templates includieren- ich hatte eher in die Richtung gedacht.

Reply:
Auch da müssen dann die entsprechenden Variablen gesetzt werden. Der Code bleibt imho fast identisch.

Reply:
Erledigt, danke für den Hint.